02_configure_host.sh

```diff
@@ -154,18 +154,52 @@ if [ "$MANAGE_INT_BRIDGE" == "y" ]; then
                grep -q BOOTPROTO /etc/sysconfig/network-scripts/ifcfg-${BAREMETAL_NETWORK_NAME} || (echo -e "\nBOOTPROTO=dhcp\n" | sudo tee -a /etc/sysconfig/network-scripts/ifcfg-${BAREMETAL_NETWORK_NAME})
            fi
        fi
-        sudo systemctl restart network
+       sudo systemctl restart network
     fi
 fi
 
 # If there were modifications to the /etc/sysconfig/network-scripts/ifcfg-*
 # files, it is required to enable the network service
-if [ "$MANAGE_INT_BRIDGE" == "y" ] || [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
+if [ "$MANAGE_INT_BRIDGE" == "y" ] ; then
   sudo systemctl enable network
 fi
 
+# VLAN interface in the provisioning bridge
+if [ ! -z "${PROVISIONING_VLAN}" ] ; then
+    VLAN_INT=${PROVISIONING_NETWORK_NAME}.${PROVISIONING_VLAN}
+    sudo cp /var/lib/libvirt/dnsmasq/${BAREMETAL_NETWORK_NAME}.conf /etc/NetworkManager/dnsmasq.d/${VLAN_INT}.conf
+    sudo sed -i "s/${BAREMETAL_NETWORK_NAME}/${VLAN_INT}/" /etc/NetworkManager/dnsmasq.d/${VLAN_INT}.conf
+    sudo sed -i "/^bind-dynamic/d" /etc/NetworkManager/dnsmasq.d/${VLAN_INT}.conf
+    sudo sed -i 's#/var/lib/libvirt/dnsmasq/#/etc/NetworkManager/dnsmasq-shared.d/#' /etc/NetworkManager/dnsmasq.d/${VLAN_INT}.conf
+    sudo cp /var/lib/libvirt/dnsmasq/${BAREMETAL_NETWORK_NAME}.hostsfile /etc/NetworkManager/dnsmasq-shared.d/${VLAN_INT}.hostsfile
+    sudo cp /var/lib/libvirt/dnsmasq/${BAREMETAL_NETWORK_NAME}.addnhosts /etc/NetworkManager/dnsmasq-shared.d/${VLAN_INT}.addnhosts
+
+    # Remove the baremetal libvirt network
+    sudo virsh net-destroy ${BAREMETAL_NETWORK_NAME}
+    sudo virsh net-undefine ${BAREMETAL_NETWORK_NAME}
```

Comment:
I was going to suggest that we modify the variables passed to the metal3-dev-env to skip creating the baremetal network, but actually this is probably nicer, so we can copy the dnsmasq config and ensure it's exactly the same in both cases :)

Reply:
Yes. This is intentional. I wanted to keep changes local to this repo for this change. But we will need to figure something out when we introduce bonds.

Reply:
I think if we reconfigure the VM definition as described below, we can just use the existing two nics for a bonded configuration, but ensure both nics are connected to the provisioning network/bridge?

Reply:
I tested this and if we remove the baremetal network, we need to find a different way to do https://github.com/openshift-metal3/dev-scripts/blob/master/utils.sh#L157